src/mantarray_waveform_analysis/peak_detection.py

```diff
@@ -251,13 +271,61 @@ def data_metrics(
                     WIDTH_UUID: widths[i],
                     AUC_UUID: auc_per_twitch[i],
                     TWITCH_FREQUENCY_UUID: twitch_frequencies[i],
+                    CONTRACTION_VELOCITY_UUID: contraction_velocity[i],
+                    RELAXATION_VELOCITY_UUID: relaxation_velocity[i],
                 }
             }
         )
 
     return main_twitch_dict, aggregate_dict
 
 
+def calculate_twitch_velocity(
+    twitch_indices: NDArray[int],
+    filtered_data: NDArray[(2, Any), int],
+    widths: List[Dict[int, Dict[UUID, Union[Tuple[int, int], int]]]],
+    is_contraction: bool,
+) -> NDArray[Any]:
+    """Find the velocity for each twitch.
+
+    Args:
+        twitch_indices: a dictionary in which the key is an integer representing the time points of all the peaks of interest and the value is an inner dictionary with various UUID of prior/subsequent peaks and valleys and their index values.
+        filtered_data: a 2D array (time vs value) of the data
+        widths: a list of dictionaries where the first key is the percentage of the way down to the nearby valleys, the second key is a UUID representing either the value of the width, or the rising or falling coordinates. The final value is either an int (for value) or a tuple of ints for the x/y coordinates
+        is_contraction: a boolean indicating if twitch velocities to be calculating are for the twitch contraction or relaxation
+
+    Returns:
+        an array of integers that are the velocities of each twitch
+    """
+    list_of_twitch_indices = list(twitch_indices.keys())
+    num_twitches = len(list_of_twitch_indices)
+    coord_type = WIDTH_RISING_COORDS_UUID
+    if not is_contraction:
+        coord_type = WIDTH_FALLING_COORDS_UUID
+
+    twitch_base = TWITCH_WIDTH_PERCENTS[-1]
```

Comment:
it's not calculated from the top to the base---let me know if this isn't clear in the diagram in the SDK documentation, but it's from 10% to 90%

Reply:
Ah, I see `TWITCH_WIDTH_PERCENTS` does currently start at 10 and go to 90.  

However, I don't think we want to rely in this function on the current circumstance that the beginning and end of TWITCH_WIDTH_PERCENTS happens to match the definition of what we're looking for for contraction/relaxation velocities.

One potential solution could be to create a new constant, something like `TWITCH_WIDTH_INDEX_OF_CONTRACTION_VELOCITY_START = TWITCH_WIDTH_PERCENTS.index(10)`. The `.index` function will find the value `10` in the array, no matter if in the future we decide to have the array start at 5, or go up by increments of 2 instead of increments of 5, etc.  

You may be able to think of a better name, but the goal would be that we decouple the scientific concept of "contraction velocities are measured from the 10% point" from the implementation detail in our code that currently we calculate the twitch width at 16 different points on the curve, the first of which happens to be `10`.